rosidl_generator_c/src/string_functions.c

```diff
@@ -101,13 +100,6 @@ bool
 rosidl_generator_c__String__assign(
   rosidl_generator_c__String * str, const char * value)
 {
-  if (!str) {
-    return false;
-  }
-  // a NULL value is not valid
-  if (!value) {
-    return false;
-  }
   return rosidl_generator_c__String__assignn(
     str, value, strlen(value));
```

Comment:
That's checked in `rosidl_geneerator_c__String__assignn`, so it's redundant to check it again here.